mm2src/coins/eth.rs

```diff
@@ -2121,7 +1969,18 @@ impl MarketCoinOps for EthCoin {
                 let uncompressed_without_prefix = hex::encode(key_pair.public());
                 Ok(format!("04{}", uncompressed_without_prefix))
             },
-            EthPrivKeyPolicy::Trezor => MmError::err(UnexpectedDerivationMethod::Trezor),
+            EthPrivKeyPolicy::Trezor => {
+                let enabled_address = self
+                    .deref()
+                    .derivation_method
+                    .hd_wallet()
+                    .ok_or(UnexpectedDerivationMethod::ExpectedHDWallet)?
+                    .get_enabled_address()
+                    .await
+                    .ok_or_else(|| UnexpectedDerivationMethod::InternalError("no enabled address".to_owned()))?
+                    .address();
+                Ok(display_eth_address(&enabled_address))
```

Comment:
Shouldn't we return the pubkey here?

Reply:
fixed (I guess we should try to avoid returning Strings to prevent bugs like this)